src/content/doc-sdk-golang/connection-engines.mdx

```diff
@@ -42,4 +42,64 @@ When you connect with a HTTP connection (`http://`, `https://`), every request y
 
 You can extend the session duration of a token or a session by setting the `DURATION` clause when creating a new access method with the [`DEFINE ACCESS METHOD`](/docs/surrealql/statements/define/access) statement or when defining a new user with the [`DEFINE USER`](/docs/surrealql/statements/define/user) statement. 
 
-Learn more about token and session duration in our [security best practices](/docs/surrealdb/reference-guide/security-best-practices#expiration) documentation.
\ No newline at end of file
+Learn more about token and session duration in our [security best practices](/docs/surrealdb/reference-guide/security-best-practices#expiration) documentation.
+
+### Example of a secure connection
+
+```go
+
+package main
+
+import (
+	"os"
+
+	surrealdb "github.com/surrealdb/surrealdb.go"
+	"github.com/surrealdb/surrealdb.go/pkg/models"
+)
+
+const (
+	defaultURL = "ws://localhost:8000"
+)
+
+var currentURL = os.Getenv("SURREALDB_URL")
+
+func getSurrealDBURL() string {
+	if currentURL == "" {
+		return defaultURL
+	}
+	return currentURL
+}
+
+func newSurrealDBConnection(namespace, database string, tables ...string) *surrealdb.DB {
+	db, err := surrealdb.New(getSurrealDBURL())
+	if err != nil {
+		panic(err)
+	}
+
+	if err = db.Use(namespace, database); err != nil {
+		panic(err)
+	}
+
+	authData := &surrealdb.Auth{
+		Username: "root",
+		Password: "root",
+	}
+	token, err := db.SignIn(authData)
+	if err != nil {
+		panic(err)
+	}
+
+	if err = db.Authenticate(token); err != nil {
+		panic(err)
+	}
+
+	// Clean up everything in the specified database
+	for _, table := range tables {
+		if _, err = surrealdb.Delete[[]any](db, models.Table(table)); err != nil {
+			panic(err)
+		}
+	}
```

Comment:
```suggestion
```

If we plan to use this helper function consistently across examples in the documentation, should we remove these lines?
I'm a bit scared that readers will just copy-paste it and end up removing every table in their app!

Reply:
Ah fair point I'll update this example